src/components/DropDown/skill-dropdown.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, Fragment, useRef, useEffect } from "react";
+import { Menu, Transition } from "@headlessui/react";
+import SelectedOption from "@/components/Tag/tag";
+import { GoChevronUp, GoChevronDown } from "react-icons/go";
+
+// Define skills
+const options = [
+  "toys",
+  "bikes",
+  "furniture",
+  "clothing",
+  "jewellery",
+  "curtains",
+  "household items",
+  "garden tools"
+];
+
+const Dropdown: React.FC = () => {
+  const [selectedItems, setSelectedItems] = useState<string[]>([]);
+  const [isOpen, setIsOpen] = useState(false);
+  const [filter, setFilter] = useState("");
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const filteredOptions = options.filter((option) =>
+    option.toLowerCase().includes(filter.toLowerCase())
+  );
+
+  const toggleItem = (item: string) => {
+    if (selectedItems.includes(item)) {
+      setSelectedItems(selectedItems.filter((i) => i !== item));
+    } else {
+      setSelectedItems([...selectedItems, item]);
+    }
+    setFilter("");
+  };
+
+  const removeItem = (item: string) => {
+    setSelectedItems(selectedItems.filter((i) => i !== item));
+  };
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        wrapperRef.current &&
+        !wrapperRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+      }
+    };
+    if (isOpen) {
+      document.addEventListener("mousedown", handleClickOutside);
+    } else {
+      document.removeEventListener("mousedown", handleClickOutside);
+    }
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [isOpen]);
+
+  return (
+    <div ref={wrapperRef}>
+      <Menu as="div" className="relative inline-block text-left w-full">
+        <div className="flex border-2 border-gray-300 rounded-md">
+          <div className="flex flex-wrap items-center flex-grow">
+            {selectedItems.map((item) => (
+              <SelectedOption key={item} option={item} onRemove={removeItem} />
+            ))}
+            {(!selectedItems.length || isOpen) && (
+              <input
+                type="text"
+                className="flex-1 pl-1 m-1 text-sm"
+                value={filter}
+                onChange={(e) => setFilter(e.target.value)}
+                onFocus={() => setIsOpen(true)}
+                placeholder="Add Skills"
+              />
+            )}
+          </div>
+          <Menu.Button
+            className="px-4 py-2 bg-slate-50 text-primary-600 rounded-r-md"
```

Comment:
Few things:
- Make sure the input element fills the entire area
- Remove the outline when the input is clicked
- Make the rounded border consistent with the user description textarea. See what it looks like if you make the textarea rounding `md` instead of `xl` and reduce the border width

What I got from playing around a bit
```
          <div className="flex flex-wrap items-center flex-grow">
            {selectedItems.map((item) => (
              <SelectedOption key={item} option={item} onRemove={removeItem} />
            ))}
            {(!selectedItems.length || isOpen) && (
              <input
                type="text"
                className="flex-1 pl-2 h-full text-sm rounded-l-xl focus:outline-none"
                value={filter}
                onChange={(e) => setFilter(e.target.value)}
                onFocus={() => setIsOpen(true)}
                placeholder="Add Skills"
              />
            )}
          </div>
          <Menu.Button
            className="px-4 py-2 bg-slate-50 text-primary-600 rounded-r-xl"
            onClick={() => setIsOpen((prevIsOpen) => !prevIsOpen)}
          >
            {isOpen ? <GoChevronUp /> : <GoChevronDown />}
          </Menu.Button>
```
<img width="350" alt="image" src="https://github.com/codersforcauses/repair-labs/assets/19729435/e410937a-83cf-43f4-8dbf-b461ed5aa6af">

Reply:
I have made the fixes and pushed them to the branch. 
Please let me know if there are any issues. Thanks :)